sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGeneratorSuite.scala

```diff
@@ -154,7 +156,15 @@ class RandomDataGeneratorSuite extends SparkFunSuite with SQLHelper {
         val data = generator.apply()
         val catalyst = toCatalyst(data)
         val convertedBack = toScala(catalyst)
-        assert(data == convertedBack)
+        dt match {
```

Comment:
shall we update the `RandomDataGenerator`? to make sure we generate correct data for `interval year`.

Reply:
> shall we update the `RandomDataGenerator`? to make sure we generate correct data for `interval year`.

Thinking this way too,  should change `RandomDataGenerator` side since it will impact many test.

Reply:
updated